lib/json_api_controller/precondition_check.rb

```diff
@@ -0,0 +1,28 @@
+module JsonApiController
+  module PreconditionCheck
+    def precondition_check
+      if !precondition
+        raise PreconditionNotPresent, precondition_error_msg
+      elsif precondition_fails?
+        raise PreconditionFailed
+      end
+    end
+    
+    def precondition
+      case action_name
+      when "show"
```

Comment:
Are we checking the modified since header timestamp for show get requests anywhere? By the look it we're only doing this for update / delete requests.